### PR TITLE
Add test for images with different crop edge

### PR DESCRIPTION
**Pull request by kbeker**

### CODE REVIEW

File: tests/golem/verificator/test_blenderverifier.py [hunk 1]
```diff
@@ -151,6 +169,39 @@ def test_docker_error(self):
...
+            verification_data['results'].append(temp_path)
+            verification_data['resources'] = self.resources
+            verification_data['paths'] = os.path.dirname(self.resources[0])
+            verifier = BlenderVerifier(verification_data, DockerTaskThread)
+            d = verifier.start_verification()
+            sync_wait(d, self.TIMEOUT)
+
+            # Change crop coordinates for next image verification
+            y_crop_cord += 30
+            y_crop_float_cord += 0.2
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: One more blank line needed.

File: tests/golem/verificator/test_blenderverifier.py [hunk 2]
```diff
@@ -38,64 +112,8 @@ def setUp(self):
...
-            self.subtask_info['total_tasks']
-        self.subtask_info['ctd']['extra_data']['crops'] = \
-            self.subtask_info['crops']
-        self.subtask_info['ctd']['extra_data']['path_root'] = \
-            self.subtask_info['path_root']
-        self.subtask_info['ctd']['extra_data']['entrypoint'] = \
-            self.subtask_info['entrypoint']
-        self.subtask_info['ctd']['subtask_id'] = self.subtask_info['subtask_id']
+        self.subtask_info = self._create_subtask_info()
+
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py:115]: Too mant blank lines.

File: tests/golem/verificator/test_blenderverifier.py [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+import random
 import shutil
 import time
 from unittest import mock
+from typing import List, Optional
 
+from golem_messages.message import ComputeTaskDef
 import pytest
 
+
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: No need for that line.

File: tests/golem/verificator/test_blenderverifier.py [hunk 4]
```diff
@@ -22,6 +27,75 @@
 class TestBlenderVerifier(TempDirFixture):
     TIMEOUT = 150
 
+    def _create_basic_subtask_info(
+            self,
+            resolution: Optional[List[int]] = None,
+            samples: Optional[int] = None,
+            borders_y: Optional[List[float]] = None,
+            entrypoint: Optional[str] = None,
+            outfilebasename = None,
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: Whitespaces not needed. `outfilebasename=None,`

File: tests/golem/verificator/test_blenderverifier.py [hunk 5]
```diff
@@ -22,6 +27,75 @@
...
+            use_compositing=False,
+            samples=samples if samples is not None else 35,
+            frames=[1],
+            output_format='PNG',
+            use_frames=False,
+            start_task=1,
+            total_tasks=1,
+            crops=list(
+                dict(
+                    outfilebasename=outfilebasename if outfilebasename is not None else "GolemTask_1",
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: more then 80 characters xD

File: tests/golem/verificator/test_blenderverifier.py [hunk 6]
```diff
@@ -22,6 +27,75 @@
...
+                    borders_x=[0.0, 1.0],
+                    borders_y=(
+                        borders_y if borders_y is not None else [0.0, 1.0]
+                    ),
+                )
+            ),
+            entrypoint=entrypoint if entrypoint is not None else \
+                            'python3 /golem/entrypoints/render_entrypoint.py',
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: My pycharm complains about indentations here.

File: tests/golem/verificator/test_blenderverifier.py [hunk 7]
```diff
@@ -22,6 +27,75 @@
...
+            path_root=os.path.dirname(self.resources[0]),
+            subtask_id=str(random.randint(1*10**36,9*10**36)),
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: Missing space, after comma.

File: tests/golem/verificator/test_blenderverifier.py [hunk 8]
```diff
@@ -22,6 +27,75 @@
...
+        )
+
+    def _create_subtask_info(
+            self,
+            resolution: Optional[List[int]] = None,
+            samples: Optional[int] = None,
+            borders_y: Optional[List[float]] = None,
+            entrypoint: Optional[str] = None,
+            outfilebasename = None,
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: No need for whitespace.

File: tests/golem/verificator/test_blenderverifier.py [hunk 9]
```diff
@@ -22,6 +27,75 @@
...
+                extra_data=dict(**self._create_basic_subtask_info(
+                    resolution,
+                    samples,
+                    borders_y,
+                    entrypoint,
+                    outfilebasename,
+                ))
+            ),
+            crop_window=[0.0, 1.0, borders_y[0], borders_y[1]] \
+                if borders_y is not None else [0.0, 1.0, 0.0, 1.0],
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py:93]: Same as above. My pycharm don't like indentation.

File: tests/golem/verificator/test_blenderverifier.py [hunk 10]
```diff
@@ -151,6 +169,39 @@ def test_docker_error(self):
...
+                get_golem_path(),
+                'tests/apps/blender/verification/test_data',
+                'GolemTask_10001.png',
+            ))
+        y_crop_cord = 0
+        y_crop_float_cord = 0.0
+        splited_images = {}
+        for i in range(1, 6):
+            # Split image to cropped parts
+            splited_images[f'image_part_{i}'] = result_image[y_crop_cord:y_crop_cord+30, 0:150]
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]: More then 80 characters here xD

File: tests/golem/verificator/test_blenderverifier.py [hunk 11]
```diff
@@ -151,6 +169,39 @@ def test_docker_error(self):
...
+
+            # Store images in temporary directory to load them to verification
+            temp_path = os.path.join(self.tempdir, f'GolemTask_1000{i}.png')
+            cv2.imwrite(temp_path, splited_images[f'image_part_{i}'])
+
+            # Create clear verification_data for every crop image
+            verification_data = {}
```
Comment (Jakub89) [inline tests/golem/verificator/test_blenderverifier.py]:
```
This dictionary creation could be rewritten as a dictionary literal.
Inspection info: This inspection detects situations when dictionary creation could be rewritten with dictionary literal.
```
Whatever that means.

**Review by Jakub89 — COMMENTED** [review]
Just some styling issues.

File: tests/golem/verificator/test_blenderverifier.py (hunk 1 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Hmm, I can't see where this line is needed :D

File: tests/golem/verificator/test_blenderverifier.py (hunk 2 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py:115]: I think done, but Pycharm says something else than linter :D

File: tests/golem/verificator/test_blenderverifier.py (hunk 3 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Done

File: tests/golem/verificator/test_blenderverifier.py (hunk 4 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Done

File: tests/golem/verificator/test_blenderverifier.py (hunk 5 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: SIUP DISABLE

File: tests/golem/verificator/test_blenderverifier.py (hunk 7 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Donee

File: tests/golem/verificator/test_blenderverifier.py (hunk 6 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: It is complicated XD because pycharm stops complaining when line looks like this:
```python
outfilebasename=outfilebasename if 
outfilebasename is not None else "GolemTask_1",
``` 

Do we want our code look like this? :D

File: tests/golem/verificator/test_blenderverifier.py (hunk 8 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Done

File: tests/golem/verificator/test_blenderverifier.py (hunk 9 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py:93]: Same as https://github.com/golemfactory/golem/pull/4045#discussion_r269173188 :D

File: tests/golem/verificator/test_blenderverifier.py (hunk 10 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Done :D

File: tests/golem/verificator/test_blenderverifier.py (hunk 6 above)
Comment (Jakub89) [reply tests/golem/verificator/test_blenderverifier.py]: Don't know, we can wait for @dybi to help decide.

File: tests/golem/verificator/test_blenderverifier.py (hunk 11 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: I changed to this:
```python 
verification_data = dict(
                subtask_info=self._create_subtask_info(
                    borders_y=[
                        0.8 - y_crop_float_cord, 1.0 - y_crop_float_cord
                    ],
                    outfilebasename=f'GolemTask_{i}'
                ),
                results=[temp_path],
                resources=self.resources,
                paths=os.path.dirname(self.resources[0]),
            )
``` 

and I think that this warning should gone :D

Comment (Jakub89) [reply tests/golem/verificator/test_blenderverifier.py]: Works

**Review by Jakub89 — APPROVED** [review]
tenor

File: tests/golem/verificator/test_blenderverifier.py [hunk 12]
```diff
@@ -151,6 +167,44 @@ def test_docker_error(self):
...
+                results=[temp_path],
+                resources=self.resources,
+                paths=os.path.dirname(self.resources[0]),
+            )
+            verifier = BlenderVerifier(verification_data, DockerTaskThread)
+            d = verifier.start_verification()
+            sync_wait(d, self.TIMEOUT)
+
+            # Change crop coordinates for next image verification
+            y_crop_cord += 30
```
Comment (dybi) [inline tests/golem/verificator/test_blenderverifier.py]: please make it a variable, ie. `y_crop_cord_step` or sth similar

File: tests/golem/verificator/test_blenderverifier.py [hunk 13]
```diff
@@ -151,6 +167,44 @@ def test_docker_error(self):
...
+            y_crop_float_cord += 0.2
```
Comment (dybi) [inline tests/golem/verificator/test_blenderverifier.py]: same as above

File: tests/golem/verificator/test_blenderverifier.py (hunk 12 above)
Comment (dybi) [inline tests/golem/verificator/test_blenderverifier.py]: BTW: the two values below seem to be updated outside `for` loop, thus all 5 `splited_images` have the same coordinates

**Review by dybi — COMMENTED** [review]
Please clarify

File: tests/golem/verificator/test_blenderverifier.py (hunk 12 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Done

File: tests/golem/verificator/test_blenderverifier.py (hunk 13 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: Done

File: tests/golem/verificator/test_blenderverifier.py (hunk 12 above)
Comment (kbeker) [reply tests/golem/verificator/test_blenderverifier.py]: But it is as we discussed :)

**Review by dybi — APPROVED** [review]
judge_approves
